README (#45)
bow-swift/nef: README.md

</p>

`nef`, short for [Nefertiti](https://en.wikipedia.org/wiki/Nefertiti), mother of Ankhesenamun, is a command line tool to ease the creation of documentation in the form of Swift Playgrounds. It provides compile-time verification of documentation and exports it in Markdown format that can be consumed by [Jekyll](https://jekyllrb.com/) to generate websites.

`nef` is inspired by [ΛNK](https://github.com/arrow-kt/ank) for Kotlin and [tut](https://github.com/tpolecat/tut) for Scala.

## 📥 Installation

> `nef` can be installed using [Homebrew](https://brew.sh). `nef` needs Xcode and [Cocoapods](https://cocoapods.org) as dependencies. It will warn you if there is a missing dependency and provide guidance to install it.

Reviewer: ```suggestion
`nef` can be installed using [Homebrew](https://brew.sh). `nef` needs Xcode and [Cocoapods](https://cocoapods.org) as dependencies. It will warn you if there is a missing dependency and will provide guidance to install it.
```

Not sure if it's better, but it's my feeling

<p align="center">
<img src="assets/nef-installation.gif">
</p>

## 🌟 Features

`nef` highlights the following features:

💡 Eases the creation of Swift Playgrounds with support for third party libraries.

💡 Compiles Swift Playgrounds with support for third party libraries from the command line.

💡 Generates Markdown project from Swift Playground.

💡 Generates Markdown files that can be consumed from Jekyll to create a microsite.

&nbsp;

### 📃 Creating a Swift Playground

Swift Playgrounds are a nice tool for prototyping and trying new concepts. However, third party libraries support is a bit cumbersome to add. One of the goals of `nef` is to make easier the creation of a Swift Playground with support for one or more libraries.

By default, `nef` can create a Swift Playground with support for [Bow](http://bow-swift.io), the Functional Programming companion library for Swift.

```bash
➜ nef playground
```

It will create an Xcode project with support for the latest available version of Bow, named `BowPlayground`. If you open this Xcode project, you will have a Swift Playground where you can import Bow or any of its modules, and start trying some of its features. 

&nbsp;

> Besides this, you can select any different `Bow version` or `Bow's branch`, even a `third-party` dependency.

Reviewer: ```suggestion
Besides this, you can select any different `Bow` version or branch, even a third-party dependency.
```

//: ## This is a single line heading 2
```

> It makes Swift Playgrounds very proper to write an article with compilable examples. The command `nef` provides to generate the Markdown files is:

Reviewer: ```suggestion
It makes Swift Playgrounds the proper tool to write an article with compilable examples. The command provided by `nef` to generate the Markdown files is:
```
